src/tools/tidy/src/lib.rs

```diff
@@ -26,18 +30,27 @@ macro_rules! t {
 }
 
 macro_rules! tidy_error {
-    ($bad:expr, $fmt:expr) => ({
-        *$bad = true;
-        eprint!("tidy error: ");
-        eprintln!($fmt);
-    });
-    ($bad:expr, $fmt:expr, $($arg:tt)*) => ({
-        *$bad = true;
-        eprint!("tidy error: ");
-        eprintln!($fmt, $($arg)*);
+    ($bad:expr, $($fmt:tt)*) => ({
+        $crate::tidy_error($bad, format_args!($($fmt)*)).expect("failed to output error");
     });
 }
 
+fn tidy_error(bad: &mut bool, args: impl Display) -> std::io::Result<()> {
+    use std::io::Write;
+    use termcolor::{Color, ColorChoice, ColorSpec, StandardStream};
+
+    *bad = true;
+
+    let mut stderr = StandardStream::stdout(ColorChoice::Auto);
+    stderr.set_color(ColorSpec::new().set_fg(Some(Color::Red)))?;
+
+    write!(&mut stderr, "tidy error")?;
+    stderr.set_color(&ColorSpec::new())?;
+
+    write!(&mut stderr, ": {args}")?;
```

Comment:
needs to be `writeln`
```
tidy check
* 633 error codes
* highest error code: E0791
tidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0760.md: leading newlinetidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0760.md: too many trailing newlines (2)* 393 features
some tidy checks failed
tidy error: `/home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0760.md` doesn't use its own error code in compile_fail exampleBuild completed unsuccessfully in 0:00:10
```
```suggestion
    writeln!(&mut stderr, ": {args}")?;
```